R/GatingSet2flowJo.R

```diff
@@ -204,16 +204,39 @@ transformationNode <- function(gh, matInfo){
                 }
 
                 if(trans.type == "flowJo_biexp"){
-                  param <-  attr(func,"parameters")
-                  transNode <- xmlNode("biex"
-                                      , namespace = "transforms"
-                                      , attrs = c("transforms:length" = param[["channelRange"]]
-                                                  , "transforms:maxRange" = param[["maxValue"]]
-                                                  , "transforms:neg" = param[["neg"]]
-                                                  , "transforms:width" = param[["widthBasis"]]
-                                                  , "transforms:pos" = param[["pos"]]
-                                                  )
-                                      )
+
+                  if(!is.null(Scaling)){
+                  # Added scaling adjustment
+                    maxRange <- param[["maxValue"]]
+                    pos.decades <- param[["pos"]]
+
+                    param.loc <- which(gsub("Comp-", "", chnl) == names(Scaling))
+                    if (length(param.loc) == 1){
+                      maxRange <- Scaling[[param.loc]][2]
+                     pos.decades <- 6.5
```

Comment:
why hardcoded value?

Reply:
This part is not actually used by me anymore. It was an old version that I thought I would leave in to give an example as to what can be done. The 6.5 was working on my data and I didn't think of making it more robust at the time since I was just testing things. I think the default is 4.5 but my data was always off of the viewing area, so I set it to 6.5. You could remove all for this section if you want. But the addition of the parameter Scaling should work for all transformations.